### PR TITLE
Independence

### PR DESCRIPTION
**Pull request by idanstark42**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
-})
+
+const { correlate } = require('./lib/correlation')
+const { makeIndependent } = require('./lib/independence')
+
+const DEFAULT_OPTIONS = {
+  independent: false
+}
+
+exports.Client = (options = DEFAULT_OPTIONS) => {
```
Comment (2roy999) [inline index.js]: this is not a constructor, so change name to function name. `create` seems to me like a good option.

File: lib/independence/independent_request.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  save () {
+    IndependentRequest.storage.put(this.key, this.data)
+  }
+
+  delete () {
+    IndependentRequest.storage.delete(this.key)
+  }
+}
+
+IndependentRequest.storage = new LocalStorageNamespace('independentRequests')
```
Comment (2roy999) [inline lib/independence/independent_request.js]: It's better leave the static in a lower place (the `index.js` is the ideal, even due we aren't always get there).
I would have created another argument for the storage and pass it on the creation of the `IndependentRequest`.

File: lib/independence/independent_request.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+IndependentRequest.all = () => {
```
Comment (2roy999) [inline lib/independence/independent_request.js]: why not static

File: lib/independence/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const Promise = require('bluebird')
```
Comment (2roy999) [inline lib/independence/index.js]: Use native promise. Usually we would get the promise as an option so we won't enforce as a package the promise library, but Axios has a disadvantage that it's using only the ES6 promise anyway so let's enforce that as the lesser of two evils. (probably we will need to remove the eslint promise rule.

File: lib/independence/local_storage_namespace.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+class LocalStorageNamespace {
+  constructor (namespace, seperator = '_') {
+    this._namespace = namespace
+    this._seperator = seperator
+  }
+
+  exists (key) {
+    return window.localStorage[this._fullKey(key)] !== undefined
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: I think it's would be better if we use the functions `setItem`, `getItem` and so on.

File: lib/independence/local_storage_namespace.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  delete (key) {
+    delete window.localStorage[this._fullKey(key)]
+  }
+
+  clear () {
+    Object.keys(this.all()).forEach(key => this.delete(key))
+  }
+
+  _partialKey (fullKey) {
+    return fullKey.replace(new RegExp(`^${this._namespace}${this._seperator}`), '')
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: I think it's more clear if we use `fullKey.substring(this_namespace.length + this._seperator.length)`

File: lib/independence/local_storage_namespace.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    return window.localStorage[this._fullKey(key)] !== undefined
+  }
+
+  get (key) {
+    return this._getValueOfFullKey(this._fullKey(key))
+  }
+
+  all () {
+    return Object.keys(window.localStorage)
+      .filter(key => key.split(this._seperator)[0] === this._namespace)
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: you are actually using split twice:
1. here
2. in `_partialKey`
*maybe* do a map to turn the keys to splitted key (arrays of the string) then filter then reduce

File: lib/independence/local_storage_namespace.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  delete (key) {
+    delete window.localStorage[this._fullKey(key)]
+  }
+
+  clear () {
+    Object.keys(this.all()).forEach(key => this.delete(key))
+  }
+
+  _partialKey (fullKey) {
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: i think `shortKey` is a better name

File: lib/independence/local_storage_namespace.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
+class LocalStorageNamespace {
+  constructor (namespace, seperator = '_') {
+    this._namespace = namespace
+    this._seperator = seperator
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js:4]: I think we should add `this._storage = window.localStorage`. I even consider to get the storage from the constructor so it will be flexible with where the data is saved.

File: lib/independence/local_storage_namespace.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
+class LocalStorageNamespace {
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js:1]: We might consider to use the same API as [Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Storage)

File: test/lib/independence/independent_request.test.js [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    clock = sinon.useFakeTimers(MOCK_DATE)
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+    clock.restore()
+  })
+
+  it('created a storage with the key `independentRequests`', () => {
+    expect(getmockStorage).to.have.been.called.with('independentRequests')
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: test is missing an act section (The idea is that the test structure should be 3 phases Arrange-Act-Assert).
If you keep the storage static, bring the `proxyquire` into the `beforeEach` or the it clause.

File: test/lib/independence/independent_request.test.js [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+
+    return request.send(mockAdapter).then(() => {
+      expect(mockAdapter).to.have.been.called.with(data)
+    })
+  })
+
+  it('saves the request if it fails with status 500', () => {
+    const error = new Error('some error')
+    error.status = 500
+    const mockAdapter = () => Promise.resolve().then(() => { throw error })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: use `Promise.reject(error)`

File: test/lib/independence/independent_request.test.js [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+    const key = 'key'
+    const data = { }
+    const request = new IndependentRequest(data, key)
+
+    return request.send(mockAdapter)
+      .then(() => { expect(false).to.equal(true) })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: 1. remember that assertions uses exception. meaning this line is throwing exception that is going to be swallowed by  the next `catch` in the promise chain. use the less common form of `then`  which is `.then(regularCallback, ErrorCallback)` or use `chai-as-promised` and make an assertion that a promise will be rejected.
3. isn't this test check 2 things: a. the request fails, b. the data have been saved.
2. use `except.fail()` instead of the current assertation

File: test/lib/independence/independent_request.test.js [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+      .catch(() => {
+        expect(mockStorage.get(key)).to.equal(data)
+      })
+  })
+
+  it('saves the request if it fails with status 0', () => {
+    const error = new Error('some error')
+    error.status = 0
+    const mockAdapter = () => Promise.resolve().then(() => { throw error })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: use `Promise.reject`

File: test/lib/independence/independent_request.test.js [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+    const key = 'key'
+    const data = { }
+    const request = new IndependentRequest(data, key)
+
+    return request.send(mockAdapter)
+      .then(() => { expect(false).to.equal(true) })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: see comment above (with the 3 items)

File: test/lib/independence/independent_request.test.js [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+  })
+
+  it('saves the request if it is successful with status 0', () => {
+    const mockAdapter = () => Promise.resolve({ status: 0 })
+    const key = 'key'
+    const data = { }
+    const request = new IndependentRequest(data, key)
+
+    return request.send(mockAdapter)
+      .then(() => { expect(false).to.equal(true) })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: Is this the right behavior? why is it soppose to fail if the adapter return resolved promise?

if I'm worng see comment above (with the 3 items)

File: test/lib/independence/independent_request.test.js [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+      .catch(() => {
+        expect(mockStorage.get(key)).to.equal(data)
+      })
+  })
+
+  it('does not save the request if it fails with status 400', () => {
+    const error = new Error('some error')
+    error.status = 400
+    const mockAdapter = () => Promise.resolve().then(() => { throw error })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: use `Promise.reject`

File: test/lib/independence/independent_request.test.js [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+    const key = 'key'
+    const data = { }
+    const request = new IndependentRequest(data, key)
+
+    return request.send(mockAdapter)
+      .then(() => { expect(false).to.equal(true) })
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: see comment above (with the 3 items)

File: test/lib/independence/independent_request.test.js [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+  it('all returns an independent request for each entry in the storage', () => {
+    mockStorage._data = {
+      key1: { field1: 'value', field2: 'another value' },
+      key2: { anotherField: 12341, field: 0.524 },
+      key3: { field1: 'value3', field2: 'another value1' },
+      key4: { field1: true, field2: false }
+    }
+    const all = IndependentRequest.all()
+
+    expect(all.length).to.eql(Object.keys(mockStorage._data).length)
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: I hate the `eql` shorthand, it's hiding what it's doing. use `deep.equal`

File: test/lib/independence/index.test.js [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+        retryedPendingRequests()
+      })
+  })
+
+  it('does not retry all pending requests on failure', () => {
+    const error = new Error('some error')
+    error.status = 500
+    requestMock.send = () => Promise.resolve().then(() => { throw error })
+    return client.get()
+      .then(() => { expect(false).to.equal(true) })
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: see comment in `independent_request.test.js` (with the 3 items)

File: test/lib/independence/index.test.js [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+    return client.get()
+      .then(() => {
+        retryedPendingRequests()
+      })
+  })
+
+  it('does not retry all pending requests on failure', () => {
+    const error = new Error('some error')
+    error.status = 500
+    requestMock.send = () => Promise.resolve().then(() => { throw error })
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: use `Promise.reject`

File: test/lib/independence/index.test.js [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+  })
+}
+
+describe('independence', () => {
+  let client
+  let clock
+
+  beforeEach(() => {
+    mockAdapter = () => Promise.resolve(mockResponse)
+    pendingRequests = [{
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: either start each object on it's own line or write each object in one line (which then you will need to start it on it's own line anyway)

File: test/lib/independence/index.test.js [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    requestMock.send = () => Promise.resolve().then(() => { throw error })
+    return client.get()
+      .then(() => { expect(false).to.equal(true) })
+      .catch(() => {
+        didNotRetryPendingRequests()
+      })
+  })
+
+  it('sets an interval to retry pending requests every 5 seconds', () => {
+    expect(client.interval).to.not.equal(undefined)
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: better would be `expect(client.interval).to.exists`

File: test/lib/independence/independent_request.test.js [hunk 24]
```diff
@@ -0,0 +1,146 @@
...
+    all.forEach((request, index) => {
+      expect(request instanceof IndependentRequest).to.equal(true)
```
Comment (2roy999) [inline test/lib/independence/independent_request.test.js]: better `expect(request).to.be.an.instanceof(IndependentRequest)`

File: test/lib/independence/local_storage_namespace.test.js [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+
+  afterEach(() => {
+    window.localStorage.clear()
+  })
+
+  it('exists returns true if the key exists', () => {
+    const key = 'key'
+    const value = 'value'
+    storage.set(key, value)
+    expect(storage.exists(key)).to.equal(true)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `except(....).to.be.true`

File: test/lib/independence/local_storage_namespace.test.js [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+  })
+
+  it('exists returns false if the key does not exist', () => {
+    const key = 'key'
+    expect(storage.exists(key)).to.equal(false)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `excpect(...).to.be.false`

File: test/lib/independence/local_storage_namespace.test.js [hunk 27]
```diff
@@ -0,0 +1,92 @@
...
+  })
+
+  it('get returns the parsed object if it exists', () => {
+    const key = 'key'
+    const value = { field: 'value', integer: 1, boolean: true }
+    storage.set(key, value)
+    expect(storage.get(key)).to.eql(value)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `deep.equal`

File: test/lib/independence/local_storage_namespace.test.js [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+  })
+
+  it('get returns undefined if it does not exists', () => {
+    const key = 'key'
+    expect(storage.get(key)).to.equal(undefined)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `expect(...).to.be.undefined`

File: test/lib/independence/local_storage_namespace.test.js [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+
+  it('all returns all the keys and values set', () => {
+    const all = {
+      key1: 'value1',
+      key2: { field1: 'value1', field2: 1000 },
+      key3: 301,
+      key4: true
+    }
+    Object.entries(all).forEach(([key, value]) => storage.set(key, value))
+    expect(storage.all()).to.eql(all)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `deep.equal`

File: test/lib/independence/local_storage_namespace.test.js [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+    expect(() => storage.set(key, 'another value')).to.throw()
+  })
+
+  it('delete removes the key from the localstorage', () => {
+    const key = 'key'
+    const value = 'value'
+    storage.set(key, value)
+    expect(window.localStorage[`${STORAGE_PREFIX}_${key}`]).to.equal(JSON.stringify(value))
+    storage.delete(key)
+    expect(window.localStorage[`${STORAGE_PREFIX}_${key}`]).to.equal(undefined)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: use `to.be.undefined`

File: test/lib/independence/local_storage_namespace.test.js [hunk 31]
```diff
@@ -0,0 +1,92 @@
...
+  it('clear removes all namespace keys from the localstorage', () => {
+    const all = {
+      key1: 'value1',
+      key2: { field1: 'value1', field2: 1000 },
+      key3: 301,
+      key4: true
+    }
+    Object.entries(all).forEach(([key, value]) => storage.set(key, value))
+    storage.clear()
+    expect(Object.keys(storage.all()).length).to.equal(0)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: better `expect(...).to.be.empty`

File: test/lib/independence/local_storage_namespace.test.js [hunk 32]
```diff
@@ -0,0 +1,92 @@
...
+    expect(Object.keys(window.localStorage).filter(key => key.startWith(STORAGE_PREFIX)).length).to.equal(0)
```
Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: better `expect(...).to.be.empty`

Comment (2roy999) [inline test/lib/independence/local_storage_namespace.test.js]: Why do you need to filter the `localStorage`? all the keys are of the same namespace which is `STORAGE_PREFIX`

File: test/lib/independence/index.test.js [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+
+const { makeIndependent } = proxyquire('../../../lib/independence', {
+  './independent_request': {
+    IndependentRequest
+  }
+})
+
+const expect = chai.expect
+
+function retryedPendingRequests () {
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: I think a better name would be with except format such as `exceptPendingRequestsToBeenRetryed`. This will infer that the function is doing some complex assertion and also would be more clear what the function is checking.

File: test/lib/independence/index.test.js [hunk 34]
```diff
@@ -0,0 +1,105 @@
...
+  expect(IndependentRequest.all).to.have.been.called()
+  pendingRequests.forEach(request => {
+    expect(request.send).to.have.been.called()
+  })
+}
+
+function didNotRetryPendingRequests () {
```
Comment (2roy999) [inline test/lib/independence/index.test.js]: as above

File: test/lib/correlation.test.js [hunk 35]
```diff
@@ -0,0 +1,49 @@
...
+
+describe('correlation', () => {
+  let client
+  let lastRequest
+
+  beforeEach(() => {
+    getCorrelationId = chai.spy(() => correlationId)
+    mockAdapter = () => Promise.resolve(mockResponse)
+    client = axios.create({ adapter: mockAdapter })
+    correlate(client)
```
Comment (2roy999) [inline test/lib/correlation.test.js:35]: Consider mocking the whole axios client

File: lib/independence/local_storage_namespace.js (hunk 10 above)
Comment (idanstark42) [reply lib/independence/local_storage_namespace.js:1]: Just add the word `item` to the end of each function? Seems redundent. I don't like extra words where you don't need them.

File: test/lib/independence/independent_request.test.js (hunk 12 above)
Comment (idanstark42) [reply test/lib/independence/independent_request.test.js]: The linter throws an error: `prefer-promise-reject-errors: Expected the Promise reject reason to be an Error.`

File: test/lib/independence/independent_request.test.js (hunk 13 above)
Comment (idanstark42) [reply test/lib/independence/independent_request.test.js]: About No. 2: the tests checks that `if` the request fails, the data has been saved. The failing of the request should be a part of the `act`. I'm asserting it here, so that someone doesn't accedentily change it so the request would succeed, and then the test would still work.
I'll fix the other two

File: lib/independence/local_storage_namespace.js (hunk 10 above)
Comment (2roy999) [reply lib/independence/local_storage_namespace.js]: Yes but this kind of thinking should have done by the people who create the Storage API.

File: test/lib/independence/independent_request.test.js (hunk 12 above)
Comment (2roy999) [reply test/lib/independence/independent_request.test.js]: oh, it's looking for the `new Error` which is fine because we don't really need the error variable so just wrtie `Promise.rejcet(new Error('some error'))`.
Also if the eslint have a problem with something that create something weird use a suppress comment:
```js
// eslint-disable-next-line prefer-promise-reject-errors
Promise.reject(error)
...
// or
...
Promise.reject(error) // eslint-disable-line  prefer-promise-reject-errors
```

File: test/lib/independence/independent_request.test.js (hunk 13 above)
Comment (2roy999) [reply test/lib/independence/independent_request.test.js]: I see it as two tests because the two tests are:
1. independent request reject sending process when adapter fails
2. independent request saves the request when adapter fails

Therefore if someone changes the `send` method such that it won't fail when adapter fails the test would immediately say what is the problem.

Also for the second test, put the assertion after an empty catch method and not inside it. This way the assertion will be checked regardless of if the `send` method rejected or not. This is good because if we decide that we don't want the send method to fail when the adapter fails we only changes one test.

**Proper unit tests should fail for exactly one reason**

File: lib/independence/local_storage_namespace.js (hunk 10 above)
Comment (idanstark42) [reply lib/independence/local_storage_namespace.js:1]: I don't like it. Do you want me to implement all the API's methods? Like `key` and `length`?

File: test/lib/correlation.test.js (hunk 35 above)
Comment (idanstark42) [reply test/lib/correlation.test.js:35]: I see where you're coming from, but it seems a bit overdue for these tests. If they fail because of the unmocked axios client, they should fail.

File: lib/independence/local_storage_namespace.js (hunk 10 above)
Comment (2roy999) [reply lib/independence/local_storage_namespace.js:1]: Well yes, if we want to implement this API we should do it to the end.
But I see what you are trying to say, if you want we can leave it like that just thought it is considering

File: lib/independence/local_storage_namespace.js [hunk 36]
```diff
@@ -1,51 +1,51 @@
...
-  constructor (namespace, seperator = '_') {
+  constructor (namespace, seperator = '_', storage) {
     this._namespace = namespace
     this._seperator = seperator
+    this._storage = storage || window.localStorage
   }
 
   exists (key) {
-    return window.localStorage[this._fullKey(key)] !== undefined
+    return this._storage[this._fullKey(key)] !== undefined
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: use `getItem`

Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: Also by the specification if the key is missing it should return `null`

File: lib/independence/local_storage_namespace.js [hunk 37]
```diff
@@ -6,7 +6,7 @@ class LocalStorageNamespace {
   }
 
   exists (key) {
-    return this._storage[this._fullKey(key)] !== undefined
+    return Boolean(this._storage.getItem(this._fullKey(key)))
```
Comment (2roy999) [inline lib/independence/local_storage_namespace.js]: we might still get the private case of an empty string. I think there is no way to get around it and write
`item != undefined` (with 1 equal) will be true for values that aren't `null` or `undefined`

Comment (idanstark42) [reply lib/independence/local_storage_namespace.js]: I think the very existance of a key in the storage is enough to return `true` to `exists`.

Comment (2roy999) [reply lib/independence/local_storage_namespace.js]: The only problem with the `Boolean` version is a false negative when an empty string is stored.
you can try on the browser:
```js
localStorage.foo = ''
Boolean(localStorage.foo)
localStorage.foo != undefined
```

Comment (idanstark42) [reply lib/independence/local_storage_namespace.js]: I used `null` instead of `undefined` becaue the linter accepts it and it works the same.

File: .eslintrc.yml [hunk 38]
```diff
@@ -12,4 +12,5 @@ plugins:
 rules:
   chai-expect/no-inner-compare: warn
   chai-expect/missing-assertion: warn
-  chai-expect/terminating-properties: warn
\ No newline at end of file
+  chai-expect/terminating-properties: warn
+  promise/no-native: off
```
Comment (2roy999) [inline .eslintrc.yml]: why?

File: .eslintrc.yml [hunk 39]
```diff
@@ -12,4 +12,5 @@ plugins:
 rules:
```
Comment (2roy999) [inline .eslintrc.yml]: also use the new eslint config. it's includes chai and mocha lints

File: index.js [hunk 40]
```diff
@@ -1,9 +1,20 @@
...
-})
+
+const { correlate } = require('./lib/correlation')
+const { makeIndependent } = require('./lib/independence')
+
+const DEFAULT_OPTIONS = {
+  independent: false
+}
+
+exports.createClient = (options = DEFAULT_OPTIONS) => {
```
Comment (2roy999) [inline index.js:10]: This is not a good way to handle default options. It's works only when you have one

File: package.json [hunk 41]
```diff
@@ -1,26 +1,29 @@
...
+    "index.js",
+    "lib/**/*.js"
+  ],
   "engines": {
-    "node": ">=8.2.1"
+    "node": ">=8.0.0"
   },
   "scripts": {
-    "test": "cross-env nyc --all --reporter=text --check-coverage --lines 90 mocha --recursive",
+    "test": "cross-env nyc --all --reporter=text --check-coverage --lines 90 mocha --recursive --require mock-local-storage",
```
Comment (2roy999) [inline package.json]: split to `test`, `test:coverage`, `test:ci`

Comment (2roy999) [inline package.json]: remove cross-env (you don't use it here)

**Review by 2roy999 — APPROVED** [review]

File: index.js (hunk 40 above)
Comment (idanstark42) [reply index.js:10]: So what do you suggest to do instead? `Obejct.assign` them?

File: .eslintrc.yml (hunk 38 above)
Comment (idanstark42) [reply .eslintrc.yml]: The promise library is injected using `options`. Do you think it should just be `bluebird`, or maybe default to `bluebird`?

File: package.json (hunk 41 above)
Comment (idanstark42) [reply package.json]: Why not? I copied the scripts from `ms-logger` and it uses it there.

File: .eslintrc.yml (hunk 39 above)
Comment (idanstark42) [reply .eslintrc.yml]: Done
